src/js/pages/main.js

```diff
@@ -0,0 +1,134 @@
+import { getProducts } from '../mocks/products.js';
+import { setListener } from '../utils/dom.js';
+import { formatToBRL } from '../utils/currency.js';
+
+class Main {
+  constructor() {
+    this.products = [];
+    this.filters = {
+      category: 'all',
+      search: '',
+      price: 'asc',
+    };
+    this.renderElement = this.getRenderElement();
+
+    this.init();
+  }
+
+  init() {
+    this.setCategoryFieldListener();
+    this.setSearchFieldListener();
+    this.setPriceFieldListener();
+
+    this.products = getProducts();
+
+    this.applyFilters();
+  }
+
+  getRenderElement() {
+    return document.querySelector('#product-list');
+  }
+
+  setCategoryFieldListener() {
+    setListener(
+      '#category-field',
+      'change',
+      this.handleCategoryFilter.bind(this)
```

Comment:
Dúvida: porque esse `bind` é necessário no `this.handleCategoryFilter`?

Reply:
Perfeito!

Como o `handleCategoryFilter` é passado como callback de um `addEventListener` aqui: https://github.com/AndrezaEmiliano/bicycle-shop/pull/3/files/24c4aa141e8892a966250c2c26aab664ca963e22#diff-ec9d2f5d36ef847a2434007ef0518069a9faec54d83a95b616c1bb74a7408532R4

E o `addEventListener` muta o `this` na execução: https://developer.mozilla.org/pt-BR/docs/Web/API/Element/addEventListener#O_valor_de_this_no_manipulador. 
É necessário "fixar" o `this` para que ele referencie a classe atual.